hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -5591,7 +5591,7 @@ public ModelPart findSubPart(String name, EntityMappingType treatTargetType) {
 		}
 
 		for ( AttributeMapping attribute : declaredAttributeMappings.valueIterator() ) {
-			if ( attribute instanceof EmbeddableValuedModelPart && attribute instanceof VirtualModelPart ) {
+			if ( attribute instanceof EmbeddableValuedModelPart  ) {
```

Comment:
I don't think this change is correct. We should only look for model parts within the embeddable type if the embeddable type is virtual/embedded, so I guess that the caller of this function should rather use `findByPath()` instead.

Reply:
@beikov reverted this change and used `findByPath()` after fixing it!